FWCore/Framework/interface/SubProcess.h

```diff
@@ -73,46 +75,58 @@ namespace edm {
     void doBeginJob();
     void doEndJob();
 
-    void doEventAsync(WaitingTaskHolder iHolder, EventPrincipal const& principal);
+    void doEventAsync(WaitingTaskHolder iHolder,
+                      EventPrincipal const& principal,
+                      std::vector<std::shared_ptr<const EventSetupImpl>> const*);
```

Comment:
Why the mulitple EventSetupImpl here? the EventPrincipal from the parent is tied to just one EventSetupImpl.

Reply:
There is one EventSetupImpl per SubProcess, that is what is in the vector. The LuminosityBlockProcessingStatus object is the thing that owns that vector and holds onto it. The SubProcesses don't know about the LuminosityBlockProcessingStatus object, only EventProcessor knows about it. The functions for the SubProcesses get called in something like a chain that starts in EventProcessor which calls a function in SubProcess, that SubProcess might call the function for another SubProcess, and so on for as many SubProcesses as exist. It seemed to me the best way to pass the EventSetupImpl's around was to percolate the vector as an argument and the SubProcess knows the index into that vector already.